holoviews/operation/datashader.py

```diff
@@ -365,6 +392,39 @@ def _process(self, element, key=None):
                 layers[c] = self.p.element_type(eldata, **params)
             return NdOverlay(layers, kdims=[data.get_dimension(agg_fn.column)])
 
+    def _apply_datashader(self, dfdata, cvs_fn, agg_fn, agg_kwargs, x, y):
+        # Suppress numpy warning emitted by dask:
+        # https://github.com/dask/dask/issues/8439
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                action='ignore', message='casting datetime64',
+                category=FutureWarning
+            )
+            agg = cvs_fn(dfdata, x.name, y.name, agg_fn, **agg_kwargs)
+
+        is_where_index = ds15 and isinstance(agg_fn, ds.where) and isinstance(agg_fn.column, rd.SpecialColumn)
+        is_summary_index = isinstance(agg_fn, ds.summary) and "index" in agg
+        if is_where_index or is_summary_index:
+            if is_where_index:
+                data = agg.data
+                agg = agg.to_dataset(name="index")
+            else:  # summary index
+                data = agg.index.data
+            neg1 = data == -1
+            for col in dfdata.columns:
+                if col in agg.coords:
+                    continue
+                val = dfdata[col].values[data]
+                if val.dtype.kind == 'f':
+                    val[neg1] = np.nan
+                elif val.dtype.kind == "O":
+                    val[neg1] = "-"
```

Comment:
This line was causing me problems for a categorical 'language' column in an example I demoed - I had to work around it by appending `"-"` as a row so it would appear as a possible category value.

Is this fixed? I guess the point of this is that when there is no data to aggregate you need some sort of special 'no category' value - the equivalent of `NaN` for categories...

Reply:
I'm not totally following the issue here. You're saying it would error because `'-'` didn't exist as a category? What precisely errored?

Reply:
Looks like @Hoxbro resolved this.